sw/airborne/modules/ins/ins_ekf2.cpp

```diff
@@ -480,6 +480,40 @@ static void send_ahrs_quat(struct transport_tx *trans, struct link_device *dev)
                               &ahrs_id);
 }
 
+static void send_eternal_pose_optitrack(struct transport_tx *trans, struct link_device *dev)
+{
+
+  float sample_temp[11];
+  #if INS_EKF2_OPTITRACK
+  sample_temp[0]  = (float) sample.time_us;
+  sample_temp[1]  = sample.pos(0) ;
+  sample_temp[2]  = sample.pos(1) ;
+  sample_temp[3]  = sample.pos(2) ; 
+  sample_temp[4]  = sample.vel(0) ;
+  sample_temp[5]  = sample.vel(1) ;              
+  sample_temp[6]  = sample.vel(2) ; 
+  sample_temp[7]  = sample.quat(0);
+  sample_temp[8]  = sample.quat(1);
+  sample_temp[9]  = sample.quat(2); 
+  sample_temp[10] = sample.quat(3);
+  #else
```

Comment:
Yes so I have done it like that because I needed some logic to make sure that the message was not crashig in the event of the ekf2 being used without optitrack. Also somehow directly using the sample struct in the message send resulted in errors. Do you have any ideas on how to make this better?